src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -2754,35 +2752,6 @@ private DestructorDeclarationSyntax ParseDestructorDeclaration(string typeName,
             }
         }
 
-        private T CheckForBlockAndExpressionBody<T>(
-            CSharpSyntaxNode block,
-            CSharpSyntaxNode expression,
-            T syntax)
-            where T : CSharpSyntaxNode
-        {
-            if (block != null && expression != null)
-            {
-                ErrorCode code;
-                if (syntax is BaseMethodDeclarationSyntax)
```

Comment:
Note: this code was wrong, and failed to account for a LocalFunction being passed in.  The updated code handles this (and has a test to validate).